### ISSUE TITLE
No network with talos 1.7 Rock64

### ISSUE THREAD

**Issue by KevinHaeusler** [siderolabs/talos#8776]
### Description

Thanks to @rothgar  we found an issue with networking on the pine64 rock64 SBC with talos >1.7.0

The SBC does boot but does not acquire a network. Manually setting the IP via kernel argument on https://factory.talos.dev does also not work.

With the specific metal-rock64-arm64.raw talos 1.6.7 image it works without any issues. 

### Logs

Sorry for the jank recording. If there is a better way to extract logs please advise and I will do so.
https://youtu.be/3JWsPTU8luE

### Environment

- Talos version: Tested with 1.7.0 and 1.7.2
- Platform: Pine64 Rock64 4GB rev 2.0

**Comment by smira** [comment]
In Talos 1.7 the SBC support was forked out of Talos tree to overlays, so I guess something with new firmware/.../linux kernel/whatever else, but we expect community to handle issues like this one.

Core Talos doesn't have any code for SBCs now, so please consider looking into https://github.com/siderolabs/sbc-rockchip and/or specific Linux kernel config for your SBC.

Good kernel logs can be acquired using serial console (not sure about this SBC, works for Raspberry PI).

**Comment by rothgar** [comment]
Do we know what may have changed from the 1.6 build vs the 1.7 overlay? I was under the impression that config didn't change, the overlay was just broken out to make management and distribution easier.

It's odd that it works in 1.6.7 without issues but not 1.7.x

**Comment by smira** [comment]
There are many changes, Linux went 6.1 to 6.6, overlay got split, it might have its own bumps, etc.

This is why we release beta versions - to get things tested out. We don't have all SBCs at hand to test them all, so if it gets broken the only option we have is to remove it.

**Comment by ehbello** [comment]
Hi all.

It took me quite some time to determine that the kernel configuration for ARM64 is missing the following options needed to enable the network stack for Rock64:

      - CONFIG_MFD_RK8XX
      - CONFIG_MFD_RK8XX_I2C
      - CONFIG_REGULATOR_RK808
      - CONFIG_RTC_RTC_DRV_RK808
      - CONFIG_DW_DMAC_CORE
      - CONFIG_DW_DMAC
      - CONFIG_DW_DMAC_PCI
      - CONFIG_COMMON_CLK_RK808

I have noticed that others may be needed for LED activity, etc., but they are outside the scope of this issue.

So, I have proceeded to create the corresponding Pull Request (https://github.com/siderolabs/pkgs/pull/972) to apply the changes in release 1.7. Together with another Pull Request (https://github.com/siderolabs/sbc-rockchip/pull/15) in sbc-rockchip to avoid duplication of MAC addresses that fixes the corresponding issue (https://github.com/siderolabs/talos/issues/7095), restore the correct operation of the network in this type of boards.

Regards.
